internal/handler/msg_createuser.go

```diff
@@ -108,65 +103,39 @@ func (h *Handler) MsgCreateUser(ctx context.Context, msg *wire.OpMsg) (*wire.OpM
 
 	common.Ignored(document, h.L, "writeConcern", "authenticationRestrictions", "comment")
 
-	defMechanisms := must.NotFail(types.NewArray("SCRAM-SHA-1", "SCRAM-SHA-256"))
+	allMechanisms := must.NotFail(types.NewArray("SCRAM-SHA-1", "SCRAM-SHA-256"))
 
-	mechanisms, err := common.GetOptionalParam(document, "mechanisms", defMechanisms)
+	mechanisms, err := common.GetOptionalParam(document, "mechanisms", allMechanisms)
 	if err != nil {
 		return nil, lazyerrors.Error(err)
 	}
 
-	var credentials *types.Document
+	if mechanisms.Len() == 0 {
+		return nil, handlererrors.NewCommandErrorMsg(
+			handlererrors.ErrBadValue,
+			"mechanisms field cannot be empty",
+		)
+	}
 
+	var password string
 	if document.Has("pwd") {
-		pwdi := must.NotFail(document.Get("pwd"))
-		pwd, ok := pwdi.(string)
+		pwd := must.NotFail(document.Get("pwd"))
```

Comment:
Let's rename this, `userPasswordV`. I heard that `pwd` stand for _print working directory_

Reply:
`passwordValue` is nice and readable IMO

Reply:
If doubt consistency is preferable. See similar usages in authenticate.go, msg_saslstart.go and other places please :pray: